File: Sports.Domain/Favorite.cs
using System;
using System.Collections.Generic;
using System.Text;
using Sports.Domain.Exceptions;
using Sports.Domain.Constants;


namespace Sports.Domain
{
    public class Favorite
    {
        public int Id { get;  set; }
        public User User { get; set; }
        public Competitor Competitor { get; set; }

        public void Validate()
        {
            CheckUserNotNull();
            CheckCompetitorNotNull();
        }

        private void CheckUserNotNull()
        {
            if (User == null)
            {
                throw new InvalidEmptyUserException(EmptyUser.EMPTY_USER);
            }
        }

        private void CheckCompetitorNotNull()
        {
            if (Competitor == null)
            {
                throw new InvalidCompetitorEmptyException(CompetitorValidation.COMPETITOR_EMPTY);
            }
        }
    }
}

File: SportsAngular/src/app/classes/favorite.ts
import { Competitor } from "./competitor";
import {User} from "./user";

export class Favorite {
    id:number;
    user:User;
    competitor:Competitor;
   
    constructor(id:number, user:User, competitor:Competitor) {
        this.id = id;
        this.user = user;
        this.competitor = competitor;
    }
}